### PR TITLE
feat(expose-loader): support alternate options to handle exporting ESM (Harmony) modules

### LINKED ISSUE #56: [Refactor] Support exposing ESM without `harmony` export
I'm using expose-loader to expose some ES6 modules, which are currently using `export default {...}`, however instead of getting the resource at my exposed variable, it's at `module.default`. Here's my require:
```javascript
require("expose-loader?Store!./store/index.js");
```

And here's my export:
```javascript
export default new Vuex.Store({
...
})
```

My module is now exposed at `Store.default`.

Is this just a side effect of ES6 modules or is there a way to expose the default directly on the variable?

### LINKED ISSUE #82: Push default export to global scope for typescript files
* Operating System: OS X
* Node Version: 10.12.0
* NPM Version: 6.5.0
* webpack Version: "4.28.0",
* expose-loader Version: "0.7.5"

This issue is for a:

- [ ] **bug**
- [x] **feature** request
- [ ] **modification** request

##### CLI Command

```bash
   webpack --mode=development
```

##### webpack.config.js

```js
module: {
        rules: [{
                test: /\.ts$/,
                include: path.join(__dirname, "src"),
                loader: "ts-loader"
            },
            {
                type: "javascript/auto",
                test: /\.json$/,
                loader: 'json-loader'
            },
            {
                test: require.resolve('./src/abc/index.ts'),
                loader: 'expose-loader?FP'
            }
        ]
    },
```

```js
 export default class abc {
    // do something
}
```

### Expected Behavior

abc is added to the global scope or under **window**

### Actual Behavior

Module parse failed: Unexpected token (11:16)
You may need an appropriate loader to handle this file type.

`export default class abc {} @ ./src/abc/index.ts-exposed 1:32-55`

### New Feature Use Case

When trying to use a class inside webpack generated bundle inside another js file (not under the scope of webpack)

### LINKED ISSUE #87: Exposing ES6 modules without the default property
Is there any news regarding exposing ES6 modules without the default property?

I searched now for quite some time, and the only thing I can find is the workaround with `module.exports = …`.

### PR DESCRIPTION
**Pull request by patrick-mcdougle**
This PR contains a:

- [ ] **bugfix**
- [x] new **feature**
- [ ] **code refactor**
- [ ] **test update** 
- [ ] **typo fix**
- [ ] **metadata update**

### Motivation / Use-Case

Fixes #56
Fixes #82 
Fixes #87 

### Breaking Changes
Old options behavior should still work, though, I don't have good test cases to prove that. Additional testing welcomed.

### Additional Info
This will be my first PR to anything node / webpack so please be nice. 😆

### CODE REVIEW

**Comment by patrick-mcdougle** [comment]
I realize I didn't follow the [commit message guidelines](https://github.com/webpack-contrib/expose-loader/blob/master/.github/CONTRIBUTING.md#-webpack-contrib-commit-conventions). I'll fix soon. 😳

**Comment by qinmingyuan** [comment]
Plea Merge this PR

File: README.md [hunk 1]
```diff
@@ -100,6 +100,27 @@ module: {
...
+  rules: {
+    test: require.resolve('./path/to/my/utility'),
+    use: [
+      loader: 'expose-loader',
+      options {
+          expose: 'Util',
+          exportKey: 'default'
+        }
+      }
+    ]
```
Comment (sheijne) [inline README.md]: There is a typo here, you're missing `{}`  inside the `use` array ;)

**Comment by patrick-mcdougle** [comment]
Ok this has been updated. There's one other chore added to this PR which is to upgrade one of the dependencies to allow node 10+ in lint-staged. Please let me know if this should be separated from this PR.

**Comment by yellow1912** [comment]
Is there any news? This is going to be very useful.

**Pull request closed without merging**